client/src/main/java/io/pravega/client/segment/impl/AsyncSegmentInputStreamImpl.java

```diff
@@ -160,8 +160,11 @@ public boolean isClosed() {
         Exceptions.checkNotClosed(closed.get(), this);
         WireCommands.ReadSegment request = new WireCommands.ReadSegment(segmentId.getScopedName(), offset, length, this.delegationToken);
 
-        return backoffSchedule.retryingOn(Exception.class)
-                .throwingOn(ConnectionClosedException.class)
+        return backoffSchedule.retryWhen(t -> {
+            Throwable ex = Exceptions.unwrap(t);
+            log.warn("Exception while reading from Segment : {}", segmentId, ex);
+            return ex instanceof Exception && !(ex instanceof ConnectionClosedException) && !(ex instanceof SegmentTruncatedException);
+        }).throwingOn(RuntimeException.class)
```

Comment:
Looking at the impl of this function in Retry: the parameter is ignored!
This seems like a bug. If it's not going to look at the parameter then we should make retryWhen() return RetryAndThrowConditionally with runtimeException directly so you don't have to type this useless line.

Reply:
https://github.com/pravega/pravega/issues/2545